src/core/issues-notes.js

```diff
@@ -146,50 +170,89 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
-async function fetchIssuesFromGithub({ githubAPI }) {
-  const issues = [];
-  const issueNumbers = [...document.querySelectorAll(".issue[data-number]")]
-    .map(elem => Number.parseInt(elem.dataset.number, 10))
-    .filter(number => number);
-  for (const issueNumber of issueNumbers) {
-    let issue; // depends on how fetching goes...
+//derives the text-color-class based on the illumination score of the labelColor
+function deriveTextColorClass(hexColor) {
+  const rgb = parseInt(hexColor, 16);
+  if (isNaN(rgb)) {
+    throw new TypeError(`Invalid hex color: ${hexColor}`);
+  }
+  const red = (rgb >> 16) & 0xff;
+  const green = (rgb >> 8) & 0xff;
+  const blue = (rgb >> 0) & 0xff;
+  const illumination = 0.2126 * red + 0.7152 * green + 0.0722 * blue;
+  return illumination > 140 ? "light" : "dark";
+}
+
+function createLabel(label) {
+  const { color, href, name } = label;
+  let textColorClass = "dark";
+  try {
+    textColorClass = deriveTextColorClass(color);
+  } catch (err) {
+    console.error(err);
+  }
+  const cssClasses = `respec-gh-label respec-label-${textColorClass}`;
+  const style = `background-color: #${color}`;
+  return hyperHTML`<a
+    class="${cssClasses}"
+    style="${style}"
+    href="${href}">${name}</a>`;
+}
+
+async function processResponse(response, issueNumber) {
+  const issue = { title: "", number: issueNumber, state: "" };
+  const jsonPromise = response.json();
+  if (!response.ok) {
+    const errJson = { message: "" };
+    try {
+      Object.assign(errJson, await jsonPromise);
+    } catch (err) {
+      errJson.message = `Error JSON parsing issue #${issueNumber} from GitHub.`;
+    } finally {
+      const msg = `Error fetching issue #${issueNumber} from GitHub. ${issue.message} (HTTP Status ${response.status}).`;
+      pub("error", msg);
+    }
+  } else {
     try {
+      Object.assign(issue, await jsonPromise);
+    } catch (err) {
+      const msg = `Error JSON parsing issue #${issueNumber} from GitHub.`;
+      pub("error", msg);
+    }
+  }
+  return [issueNumber, issue];
+}
+
+async function fetchAndStoreGithubIssues(githubAPI) {
+  const specIssues = Array.from(document.querySelectorAll(".issue[data-number]"));
+  if (specIssues.length > MAX_GITHUB_REQUESTS) {
+    const msg = `Your spec contains ${specIssues.length} Github issues, `
+      + `but GitHub only allows ${MAX_GITHUB_REQUESTS} requests. Some issues might not show up.`;
+    pub("warning", msg);
+  }
+  const issuePromises = specIssues
+    .map(elem => Number.parseInt(elem.dataset.number, 10))
+    .filter(issueNumber => issueNumber)
+    .map(async issueNumber => {
```

Comment:
this function, and processResponse should move to core/github, into a "export async function fetchIssue(number){...}"